hub/core/storage/s3/s3_mapper.py

```diff
@@ -0,0 +1,173 @@
+import posixpath
+import boto3
+import botocore
+from typing import Optional
+from collections.abc import MutableMapping
+from botocore.exceptions import ClientError
+from hub.util.exceptions import S3GetError, S3SetError, S3DeletionError, S3ListError
+
+
+class S3Mapper(MutableMapping):
```

Comment:
I'm still in favor of moving this into S3Provider and subclassing on Provider directly. Is that still happening?